lib/components/table/table.less

```diff
@@ -122,10 +122,24 @@
 
     &&__sortable {
         thead th {
-            a { //  If an anchor is used, it should appear like a normal header
+            a,
+            button { //  If an anchor is used, it should appear like a normal header
                 color: inherit;
             }
 
+            button {
+                appearance: none;
+                background-color: transparent;
+                border: 0;
+                padding: var(--su8);
+                text-align: left;
+                width: 100%;
+            }
+
+            &[data-s-table-target="column"] { // If this column is sortable, then the padding will come from the button
```

Comment:
I think there's some CSS we can apply that will address both issues. Basically, we can remove the padding override targeting the data attribute and update the button styling to something like this:

```
button {
    appearance: none;
    background-color: transparent;
    border: 0;
    display: flex;
    gap: var(--su-static4);
    font-weight: inherit;
    margin: calc(var(--_ta-th-p) * -1);
    padding: var(--_ta-th-p);
    text-align: left;
    width: calc(100% + calc(var(--_ta-th-p) * 2));
}
```

This will ensure that the sizing responds to the size modifiers.

![20230502-044643](https://user-images.githubusercontent.com/647177/235782185-8d35f42e-f301-494b-997a-559c087c07b6.gif)

I've pushed this change in https://github.com/StackExchange/Stacks/pull/1280/commits/10b76fba352b927c703eb868ed9fcb345fc655d2. Let me know if it seems reasonable to you.

PS: I honestly don't _love_ using negative margins in general, but since they're used relative to a pseudo-private custom property for this component, it seems safe.

Reply:
@dancormier You are indeed a CSS wizard. 😀